mapper-processor/src/main/java/com/datastax/oss/driver/internal/mapper/processor/entity/DefaultEntityFactory.java

```diff
@@ -436,20 +463,23 @@ private void scanFieldAnnotations(
   }
 
   private void scanMethodAnnotations(
+      Set<TypeElement> typeHierarchy,
       ExecutableElement getMethod,
-      Map<Class<? extends Annotation>, Annotation> annotations,
-      Class<? extends Annotation> exclusiveAnnotation) {
+      Map<Class<? extends Annotation>, Annotation> annotations) {
+    Class<? extends Annotation> exclusiveAnnotation = getExclusiveAnnotation(annotations);
     for (Class<? extends Annotation> annotationClass : PROPERTY_ANNOTATIONS) {
-      Annotation annotation = getMethod.getAnnotation(annotationClass);
-      if (annotation != null) {
+      Optional<? extends ResolvedAnnotation<? extends Annotation>> annotation =
+          getMethodAnnotation(annotationClass, getMethod, typeHierarchy);
+      if (annotation.isPresent()) {
         if (EXCLUSIVE_PROPERTY_ANNOTATIONS.contains(annotationClass)) {
```

Comment:
what if someone will annotate a method with 3 exclusive annotations?
We will report in the `reportMultipleAnnotationError` method only the first two, right?
Maybe we should report all of them?

Reply:
I agree that it would be a nice to have, but i see it as low value for the effort it would take.  If have some time I'll look into it.